engines/private/symbol.cpp

```diff
@@ -103,30 +109,33 @@ static Symbol *install(const Common::String &n, int t, int d, const char *s, Com
 	return sp;
 }
 
+
 /* lookup some name in some symbol table */
-Symbol *SymbolMaps::lookupName(const char *n) {
-	//debug("looking up %s", n);
-	Common::String s(n);
+Symbol *SymbolMaps::lookupRect(Common::String *n) {
+	//debug("looking rect up %s", n->c_str());
 
-	if (settings.contains(s))
-		return lookup(s, settings);
+	assert(rects.contains(*n));
+	return lookup(*n, rects);
+}
 
-	else if (variables.contains(s))
-		return lookup(s, variables);
+Symbol *SymbolMaps::lookupVariable(Common::String *n) {
+	//debug("looking variable up %s", n->c_str());
+
+	assert(variables.contains(*n));
+	return lookup(*n, variables);
+}
 
-	else if (cursors.contains(s))
-		return lookup(s, cursors);
 
-	else if (locations.contains(s))
-		return lookup(s, locations);
+/* lookup some name in some symbol table */
+Symbol *SymbolMaps::lookupName(const char *n) {
 
-	else if (rects.contains(s))
-		return lookup(s, rects);
+	Symbol *s = (Symbol *)malloc(sizeof(Symbol));
+	Common::String *name = new Common::String(n);
+	s->name = name;
```

Comment:
why creating `name` variable? you could assign directly: `s->name = new Common::String(n)`

Reply:
Done.